lib/cli/cmd-add.js

```diff
@@ -0,0 +1,23 @@
+var Q = require('q');
+var runBinary = require('./run-binary');
+
+// `donejs add app [folder]`     => `donejs-cli init [folder]`
+// `donejs add plugin [folder]`  => `donejs-cli init [folder] --type=plugin`
+// `donejs add generator [name]` => `donejs-cli init [folder] --type=generator`
+module.exports = function(type, folder, options) {
+  if (!isValidType(type)) {
+    var msg = 'Invalid [type] value `' + type + '`';
+    return Q.reject(new Error(msg));
+  }
+
+  if (type !== 'app') {
+    options.type = type;
+  }
+
+  return runBinary(['init', folder], options);
+};
+
+function isValidType(type) {
```

Comment:
It's a pet peeve of mine, too. Imagine if this would be 1000 lines of code and I'd skim through it I have no idea what this function does until I find it somewhere later on.